test/exercises/diffusion/heat_2d.chpl

```diff
@@ -1,55 +1,46 @@
-// ---- set up simulation parameters ------
-// declare configurable parameters with default values
-config const xLen = 2.0,    // length of the domain in x
-             yLen = 2.0,    // length of the domain in y
-             nx = 31,       // number of grid points in x
-             ny = 31,       // number of grid points in y
-             nt = 50,       // number of time steps
-             sigma = 0.25,  // CFL condition
-             nu = 0.05;     // viscosity
+/*
+  A 2D finite difference heat/diffusion equation solver
+
+  Computation is local (single compute node) and runs
+  the kernel in parallel using a `forall` loop
 
-// declare non-configurable parameters
-const dx: real = xLen / (nx - 1),       // grid spacing in x
-      dy: real = yLen / (ny - 1),       // grid spacing in y
-      dt: real = sigma * dx * dy / nu;  // time step size
+  Values of the `config const` variables can be modified in
+  the command line (e.g., `./heat_2D --nt=100`)
+*/
+
+// declare configurable constants with default values
+config const nx = 256,      // number of grid points in x
+             ny = 256,      // number of grid points in y
+             nt = 50,       // number of time steps
+             alpha = 0.25;  // diffusion constant
 
-// ---- set up the grid ------
 // define a 2D domain and subdomain to describe the grid and its interior
 const indices = {0..<nx, 0..<ny},
-      indicesInner = {1..<nx-1, 1..<ny-1};
+      indicesInner = {1..<nx-1, 1..<ny-1}; // equivalent to ` = indices.expand(-1)`
 
 // define a 2D array over the above domain
 var u: [indices] real;
 
 // set up initial conditions
 u = 1.0;
-u[
-  (0.5 / dx):int..<(1.0 / dx + 1):int,
-  (0.5 / dy):int..<(1.0 / dy + 1):int
-] = 2;
+u[nx/4..nx/2, ny/4..ny/2] = 2.0;
 
-// ---- run the finite difference computation ------
 // create a temporary copy of 'u' to store the previous time step
 var un = u;
 
 // iterate for 'nt' time steps
 for 1..nt {
-
-  // swap the arrays to prepare for the next time step
+  // swap arrays to prepare for next time step
   u <=> un;
 
-  // update the solution over the interior of the domain in parallel
-  forall (i, j) in indicesInner {
-    u[i, j] = un[i, j] +
-              nu * dt / dy**2 *
-                (un[i-1, j] - 2 * un[i, j] + un[i+1, j]) +
-              nu * dt / dx**2 *
-                (un[i, j-1] - 2 * un[i, j] + un[i, j+1]);
-  }
+  // compute the FD kernel in parallel
+  forall (i, j) in indicesInner do
+    u[i, j] = un[i, j] + alpha *
+      (un[i, j-1] + un[i-1, j] + un[i+1, j] + un[i, j+1] - 4 * un[i, j]);
 }
 
-// ---- print final results ------
+// print final results
 const mean = (+ reduce u) / u.size,
       stdDev = sqrt((+ reduce (u - mean)**2) / u.size);
 
-writeln((0.102424 - stdDev) < 1e-6);
+writeln(abs(0.222751 - stdDev) < 1e-6);
```

Comment:
It's just the standard deviation of the solution for this particular heat problem with the given parameters. Ideally we'd call out to some other heat solver or use a closed form solution to check correctness.

Reply:
Oh, is it specific to the particular problem size then?  The correctness size, presumably, but not the performance size?

I think this should be made a config const so that (a) it has a name and (b) we can change it for different runs.

This is also making me realize that the performance version of the test should have some way of validating the output, whether that's checking for this true/false output (where it'd presumably need to be labeled like `validation: ` and then we'd have a `verify` line in the .perfkeys file), or we'd just not print out the timing in the case that validation failed, which will cause the `time: ` line to be missing and for it to break in that way.  (or maybe something else).  That feels important to have happen before this gets merged to make sure someone doesn't break any of the versions in ways that make the performance data invalid.

Reply:
> The correctness size, presumably, but not the performance size?

That's correct. I'll make it configurable and add a `verify: ` perf-key.